sn_transfers/src/wallet/data_payments.rs

```diff
@@ -163,4 +163,67 @@ impl PaymentQuote {
             signature: vec![],
         }
     }
+
+    /// Check whether self is newer than the target quote.
+    pub fn is_newer_than(&self, other: &Self) -> bool {
+        self.timestamp > other.timestamp
+    }
+
+    /// Check against a new quote, verify whether it is a valid one from self perspective.
+    /// Returns `true` to flag the `other` quote is valid, from self perspective.
+    pub fn historical_verify(&self, other: &Self) -> bool {
+        // There is a chance that an old quote got used later than a new quote
+        let self_is_newer = self.is_newer_than(other);
+        let (old_quote, new_quote) = if self_is_newer {
+            (other, self)
+        } else {
+            (self, other)
+        };
+
+        if new_quote.quoting_metrics.live_time < old_quote.quoting_metrics.live_time {
+            info!("Claimed live_time out of sequence");
+            return false;
+        }
+
+        let old_elapsed = if let Ok(elapsed) = old_quote.timestamp.elapsed() {
+            elapsed
+        } else {
+            info!("timestamp failure");
```

Comment:
let's warn these and get some more info on the Peer that's failing logged out here I think

Reply:
there is a warning log at the place of received the failed verify,
which will log the entire quote for investigation.
don't want to abuse the log file duplicatedly, hence here just log simple for easier debugging.